File: app/src/main/java/be/digitalia/fosdem/db/converters/NonNullDateTypeConverters.java
package be.digitalia.fosdem.db.converters;

import java.util.Date;

import androidx.annotation.NonNull;
import androidx.room.TypeConverter;

public class NonNullDateTypeConverters {
	@TypeConverter
	public static Date toDate(long value) {
		return new Date(value);
	}

	@TypeConverter
	public static long fromDate(@NonNull Date value) {
		return value.getTime();
	}
}

File: app/src/main/java/be/digitalia/fosdem/db/converters/NonNullDateTypeConverters.kt
package be.digitalia.fosdem.db.converters

import androidx.room.TypeConverter
import java.util.*

object NonNullDateTypeConverters {
    @JvmStatic
    @TypeConverter
    fun toDate(value: Long): Date = Date(value)

    @JvmStatic
    @TypeConverter
    fun fromDate(value: Date): Long = value.time
}